ocamldoc/odoc_sig.ml

```diff
@@ -282,23 +282,26 @@ module Analyser =
           | Ptyp_object (fields, _) ->
             let rec f = function
               | [] -> []
-              | ({txt=""},_,_) :: _ ->
+              | Otag ({txt=""},_,_) :: _ ->
                 (* Fields with no name have been eliminated previously. *)
                 assert false
-
-              | ({txt=name}, _atts, ct) :: [] ->
+              | Otag ({txt=name}, _atts, ct) :: [] ->
                 let pos = Loc.ptyp_end ct in
                 let (_,comment_opt) = just_after_special pos pos_end in
                 [name, comment_opt]
-              | ({txt=name}, _atts, ct) :: ((_name2, _atts2, ct2) as ele2) :: q ->
+              | Otag ({txt=name}, _, ct) ::
+                  ((Oinherit ct2) as ele2) :: q
+              | Otag ({txt=name}, _, ct) ::
+                  ((Otag (_, _, ct2)) as ele2) :: q ->
                 let pos = Loc.ptyp_end ct in
                 let pos2 = Loc.ptyp_start ct2 in
                 let (_,comment_opt) = just_after_special pos pos2 in
                 (name, comment_opt) :: (f (ele2 :: q))
+              | _ :: q -> f q
```

Comment:
@objmagic, sorry for the delayed answer. I will try to have by a look by the end of the month to see if it is possible to implement documentation comment for inherited fields in a reasonable way within ocamldoc. However, I don't think that non-optimal support within ocamldoc should block integration ((inlined record had a quite imperfect support for a time, and documentation comment can still not be attached to polymorphic variants)).